dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java

```diff
@@ -1294,4 +1308,246 @@ public boolean canCreateNewVersion(Context context, Item item) throws SQLExcepti
+    //and the keys of the hashmap will be used to construct the RelationshipMetadataValue object.
+    private List<RelationshipMetadataValue> handleRelationshipTypeMetadataMapping(Context context, Item item,
+                                                HashMap<String, VirtualMetadataConfiguration> hashMaps,
+                                                                                  Item otherItem, String relationName,
+                                                                                  Integer relationshipId, int place)
+        throws SQLException {
+        List<RelationshipMetadataValue> resultingMetadataValueList = new LinkedList<>();
+        for (Map.Entry<String, VirtualMetadataConfiguration> entry : hashMaps.entrySet()) {
+            String key = entry.getKey();
+            VirtualMetadataConfiguration virtualBean = entry.getValue();
+
+            for (String value : virtualBean.getValues(context, otherItem)) {
+                RelationshipMetadataValue metadataValue = constructMetadataValue(context, key);
+                if (metadataValue != null) {
+                    metadataValue = constructResultingMetadataValue(item, value, metadataValue, relationshipId);
+                    metadataValue.setUseForPlace(virtualBean.getUseForPlace());
+                    metadataValue.setPlace(place);
+                    if (StringUtils.isNotBlank(metadataValue.getValue())) {
+                        resultingMetadataValueList.add(metadataValue);
+                    }
+                }
+            }
+        }
+        return resultingMetadataValueList;
+    }
+
+    private RelationshipMetadataValue getRelationMetadataFromOtherItem(Context context, Item otherItem,
+                                                                       String relationName,
+                                                                       Integer relationshipId) {
+        RelationshipMetadataValue metadataValue = constructMetadataValue(context,
+                                                                         MetadataSchemaEnum.RELATION
+                                                                             .getName() + "." + relationName);
+        if (metadataValue != null) {
+            metadataValue.setAuthority(Constants.VIRTUAL_AUTHORITY_PREFIX + relationshipId);
+            metadataValue.setValue(otherItem.getID().toString());
+            return metadataValue;
+        }
+        return null;
+    }
+
+    private String getEntityTypeStringFromMetadata(List<MetadataValue> list) {
+        String entityType = null;
+        for (MetadataValue mdv : list) {
+            if (StringUtils.equals(mdv.getMetadataField().getMetadataSchema().getName(),
+                                   "relationship")
+                && StringUtils.equals(mdv.getMetadataField().getElement(),
+                                      "type")) {
+
+                entityType = mdv.getValue();
+            }
```

Comment:
Trivial:  missing 'break;'

Reply:
This has been included in https://github.com/DSpace/DSpace/pull/2418 by returning when it is found